crates/cairo-profiler/src/trace_reader.rs

```diff
@@ -125,28 +125,32 @@ pub fn collect_samples_from_trace(trace: &CallTrace) -> Vec<Sample> {
     samples
 }
 
-pub struct ResourcesKeys {
-    pub builtins: HashSet<String>,
-    pub syscalls: HashSet<DeprecatedSyscallSelector>,
-}
+pub struct SampleUnits(HashSet<String>);
 
-impl ResourcesKeys {
-    pub fn measurement_types(&self, context: &mut ProfilerContext) -> Vec<ValueType> {
-        let mut value_types = vec![];
+impl SampleUnits {
+    pub fn new(mut units: HashSet<String>) -> Self {
```

Comment:
```suggestion
    pub fn from(mut units: HashSet<String>) -> Self {
```

Reply:
Disagree, it is a constructor just like `Box::new(sth)`